src/common/file.c

```diff
@@ -229,32 +354,51 @@ util_file_map_whole(const char *path)
 }
 
 /*
- * util_file_zero_whole -- zeroes the entire file
+ * util_file_zero -- zeroes the specified region of the file
  */
 int
-util_file_zero_whole(const char *path)
+util_file_zero(const char *path, off_t off, size_t len)
 {
+	LOG(3, "path \"%s\" off %ju len %zu", path, off, len);
+
 	int fd;
 	int olderrno;
 	int ret = 0;
 
-	if ((fd = os_open(path, O_RDWR)) < 0)
+	if ((fd = os_open(path, O_RDWR)) < 0) {
+		ERR("!open \"%s\"", path);
 		return -1;
+	}
 
 	ssize_t size = util_file_get_size(path);
 	if (size < 0) {
+		LOG(2, "cannot determine file length \"%s\"", path);
 		ret = -1;
 		goto out;
 	}
 
 	void *addr = util_map(fd, (size_t)size, MAP_SHARED, 0, 0);
 	if (addr == NULL) {
+		LOG(2, "failed to map entire file \"%s\"", path);
 		ret = -1;
 		goto out;
 	}
 
+	if (off > size) {
+		LOG(2, "offset beyond file length, %ju > %ju", off, size);
+		ret = -1;
+		goto out;
+	}
+
+	if ((size_t)off + len > (size_t)size) {
+		LOG(2, "requested size of write goes beyond the file length, "
```

Comment:
this should be an ASSERT and the user should never encounter this message (unless we plan on exposing some kind of zeroing command?)